zulipterminal/model.py

```diff
@@ -564,6 +565,60 @@ def update_private_message(self, msg_id: int, content: str) -> bool:
         display_error_if_present(response, self.controller)
         return response["result"] == "success"
 
+    def get_latest_message_in_topic(
+        self, stream_id: int, topic_name: str
+    ) -> Dict[str, Any]:
+        request: Dict[str, Any] = {
+            "anchor": "newest",
+            "num_before": 1,
+            "num_after": 0,
+            "narrow": [
+                {
+                    "operator": "stream",
+                    "operand": self.stream_dict[stream_id]["name"],
+                },
+                {"operator": "topic", "operand": topic_name},
+            ],
+        }
+        response = self.client.get_messages(request)
+        display_error_if_present(response, self.controller)
+        return response
+
+    def can_user_edit_topic(self) -> bool:
+        if not self.initial_data.get("realm_allow_message_editing", None):
+            self.controller.report_error(" Editing messages is disabled")
+            return False
+        elif self.initial_data.get("realm_allow_community_topic_editing", None):
+            self.controller.report_error(" Editing topic is disabled")
+            return False
+        else:
+            return True
+
+    def toggle_topic_resolved_status(self, stream_id: int, topic_name: str) -> None:
+        if self.can_user_edit_topic():
+            response = self.get_latest_message_in_topic(stream_id, topic_name)
+            if response["messages"] != []:
+                time_since_msg_sent = time.time() - response["messages"][0]["timestamp"]
+                edit_time_limit = self.initial_data[
+                    "realm_community_topic_editing_limit_seconds"
+                ]
+                # Don't allow editing topic if time-limit exceeded.
+                if time_since_msg_sent >= edit_time_limit:
+                    self.controller.report_error(
+                        " Time limit for editing topic has been exceeded."
+                    )
+                else:
+                    if topic_name.startswith(RESOLVED_TOPIC_PREFIX):
+                        topic_name = topic_name[2:]
+                    else:
+                        topic_name = RESOLVED_TOPIC_PREFIX + topic_name
+                    args = dict(
+                        message_id=response["messages"][0]["id"],
+                        topic=topic_name,
+                        propagate_mode="change_all",
+                    )
+                    self.update_stream_message(**args)
+
     def update_stream_message(
```

Comment:
This is outside your changes, but I noticed that the code splits this function from the private equivalent, and it would be good to keep them together.